src/rosdep2/lookup.py

```diff
@@ -127,8 +127,15 @@ def get_rule_for_platform(self, os_name, os_version, installer_keys, default_ins
                     # we've already checked for PACKAGE_MANAGER_KEY, so
                     # version key must be present here for data to be valid
                     # dictionary value.
-                    if os_version not in data:
+                    # if the os_version is not defined and there is no wildcard
+                    if os_version not in data and '*' not in data:
                         raise ResolutionError(rosdep_key, self.data, os_name, os_version, 'No definition of [%s] for OS version [%s]' % (rosdep_key, os_version))
+                    # if the os_version has the value None
+                    if os_version in data and data[os_version] is None:
+                        raise ResolutionError(rosdep_key, self.data, os_name, os_version, 'Null definition of [%s] for OS version [%s]' % (rosdep_key, os_version))
```

Comment:
I'm afraid `Null definition` and `No definition` (from a [few lines above](https://github.com/ros-infrastructure/rosdep/pull/573/files#diff-6c706391ac7472f4d9badd303bc707f6R132)) may sound the same to majority of users. IIUC, what "Null definition" here means is that there's no way that we get the binary of a certain package (without additional community effort etc.). Do we not want to clearly tell that to the users?

Reply:
Please suggest a specific change which in your opinion would clarify the message.

Reply:
Suggestion: `'[%s] defined as "not available" for OS version [%s]'`

Reply:
Since the PR has already been merged and released please create a new PR for the proposed change.

Reply:
#581